src/coreclr/src/jit/gentree.h

```diff
@@ -3579,9 +3629,12 @@ struct GenTreeCall final : public GenTree
     regList regArgList;
 #endif
 
-    // TODO-Throughput: Revisit this (this used to be only defined if
-    // FEATURE_FIXED_OUT_ARGS was enabled, so this makes GenTreeCall 4 bytes bigger on x86).
-    CORINFO_SIG_INFO* callSig; // Used by tail calls and to register callsites with the EE
+#ifdef DEBUG
+    // Used to register callsites with the EE
```

Comment:
I don't know why -- right now it is only registered in debug:
https://github.com/dotnet/runtime/blob/5c12e28775077cf8b585f48bc83137e931283634/src/coreclr/src/jit/emitxarch.cpp#L12665-L12671

The previous tailcall-via-help mechanism needed the call signature also, so the field was used for both purposes (registering call-sites in debug and otherwise tailcall-via-help in some cases). The new mechanism needs more information than just the signature, so I introduced a new field for it and to avoid growing the size of `gtCall` in release builds I marked the signature (which is now only used for registering call sites) as debug only.

Reply:
I'll leave this as is for now.